server/s3_bucket_metadata.cc

```diff
@@ -42,7 +43,11 @@ S3BucketMetadata::S3BucketMetadata(
   user_name = request->get_user_name();
   owner_canonical_id = request->get_canonical_id();
   user_id = request->get_user_id();
-  bucket_name = request->get_bucket_name();
+  if (bucket.empty()) {
+    bucket_name = request->get_bucket_name();
+  } else {
+    bucket_name = bucket;
+  }
```

Comment:
We would not need this. I'm missing here what is the need for passing 'bucket' parameter to constructor. Similar comment above. Can you justify this need?

Reply:
yes removed this.